acvm/src/compiler/mod.rs

```diff
@@ -26,23 +28,52 @@ pub enum CompileError {
     UnsupportedMemoryOpcode(UnsupportedMemoryOpcode),
 }
 
+fn create_acir_transformation_map(
+    acir: &Circuit,
+    acir_opcode_positions: Vec<usize>,
+) -> HashMap<OpcodeLocation, OpcodeLocation> {
```

Comment:
possibly create a type alias, since `HashMap<OpcodeLocation, OpcodeLocation>` is not the most informative type and it is used in compile

Reply:
Yeah, I opted for a full struct to avoid creating a large map for large circuits, and a method to transform old locations to the new ones